legal-api/src/legal_api/resources/business/business_tasks.py

```diff
@@ -140,56 +135,44 @@ def construct_task_list(business):  # pylint: disable=too-many-locals; only 2 ex
             tasks.append(task)
             order += 1
 
-        last_ar_date = business.last_ar_date
-        if check_agm:
-            # If this is a CO-OP, set the start date to the first day of the year, since an AR filing
-            # is available as of Jan/01
-            if last_ar_date:
-                todo_start_date = (datetime(last_ar_date.year + 1, 1, 1)).date()
-            else:
-                # If this is the first calendar year since incorporation, there is no
-                # previous ar date. Use the next anniversary year.
-                todo_start_date = (datetime(todo_start_date.year, 1, 1)).date()
+        # If this is the first calendar year since incorporation, there is no previous ar year.
+        next_ar_year = (business.last_ar_year if business.last_ar_year else business.founding_date.year) + 1
+        start_date = (business.last_ar_date if business.last_ar_date else business.founding_date).date()
 
-        # Retrieve all previous annual report filings. If there are existing AR filings, determine
-        # the latest date of filing
+        # Checking for pending ar
         annual_report_filings = Filing.get_filings_by_type(business.id, 'annualReport')
         if annual_report_filings:
-            # get last AR date from annualReportDate; if not present in json, try annualGeneralMeetingDate and
-            # finally filing date
-            last_ar_date = \
-                annual_report_filings[0].filing_json['filing']['annualReport'].get('annualReportDate', None)
-            if not last_ar_date:
-                last_ar_date = annual_report_filings[0].filing_json['filing']['annualReport']\
-                    .get('annualGeneralMeetingDate', None)
-            if not last_ar_date:
-                last_ar_date = annual_report_filings[0].filing_date
-            last_ar_date = datetime.fromisoformat(last_ar_date)
-            if check_agm:
-                todo_start_date = (datetime(last_ar_date.year+1, 1, 1)).date()
-            else:
-                todo_start_date = (last_ar_date+datedelta.YEAR).date()
+            # Consider each filing as each year and add to find next ar year
+            next_ar_year += len(annual_report_filings)
```

Comment:
if a business was founded in 2018 and they had filed their 2019+2020 annual reports (so business.last_ar_year would equal 2020), then their next_ar_year = 2020 + 2 = 2022. Am I missing something or does this logic need to change?

Reply:
`annual_report_filings` are the filings which are in not completed status, at that point `last_ar_year` will not have pending or draft filings (as per the functionality I dont see a possibility to have more than one in `annual_report_filings`). so in that case 2019 and 2020 will not be in `annual_report_filings`

Reply:
ah okay you are right. That method name 'get_filings_by_type' is a bit misleading (it was probably me way back when lol) as it doesn't return completed filings